shell/platform/linux/fl_plugin_registrar_private.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_PLUGIN_REGISTRAR_PRIVATE_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_PLUGIN_REGISTRAR_PRIVATE_H_
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_binary_messenger.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_plugin_registrar.h"
+
+G_BEGIN_DECLS
+
+/**
+ * fl_plugin_registrar_new:
+ * @view: the #FlView that is being plugged into.
+ * @messenger: the #FlBinaryMessenger to communicate with.
+ *
+ * Creates a new #FlPluginRegistrar.
+ *
+ * Returns: a new #FlPluginRegistrar.
+ */
+FlPluginRegistrar* fl_plugin_registrar_new(FlView* view,
```

Comment:
You may need to restructure this to not be a constructor argument when adding headless support to the engine, but since it's internal we can figure that out later.

Reply:
Yes, I expect this constructor to change a lot as we add new functionality.